faux-mgs/src/main.rs

```diff
@@ -643,6 +635,138 @@ impl std::fmt::Display for CfpaSlot {
     }
 }
 
+#[derive(Clone, Debug, clap::Args)]
+pub struct IgnitionSelector {
+    /// Ignition target
+    #[clap(short, long)]
+    target: Option<u8>,
+
+    /// Sled cubby (0-31)
+    #[clap(short, long, conflicts_with_all = ["target"])]
+    cubby: Option<u8>,
+
+    /// Sidecar ('local' or 'peer')
+    #[clap(
+        short, long, conflicts_with_all = ["target", "cubby"],
+        value_parser = parse_sidecar_selector
+    )]
+    sidecar: Option<SidecarSelector>,
+
+    /// PSC index (0-1)
+    #[clap(short, long, conflicts_with_all = ["target", "cubby", "sidecar"])]
+    psc: Option<u8>,
+}
+
+impl IgnitionSelector {
+    /// Decodes CLI arguments to an ignition target
+    fn get_target(&self, log: &Logger) -> Result<Option<u8>> {
+        // At this point, we assume that the various `Option` values are
+        // mutually exclusive (enforced by `clap`).
+        let t = if let Some(t) = self.target {
+            t
+        } else if let Some(c) = self.cubby {
+            // See RFD 144 § 6.1 (Switch Port Map) for this mapping
+            let t = match c {
+                0 => 14,
+                1 => 30,
+                2 => 15,
+                3 => 31,
+                4 => 13,
+                5 => 29,
+                6 => 12,
+                7 => 28,
+                8 => 10,
+                9 => 26,
+                10 => 11,
+                11 => 27,
+                12 => 9,
+                13 => 25,
+                14 => 8,
+                15 => 24,
+                16 => 4,
+                17 => 20,
+                18 => 5,
+                19 => 21,
+                20 => 7,
+                22 => 6,
+                24 => 0,
+                25 => 16,
+                26 => 1,
+                27 => 17,
+                28 => 3,
+                29 => 19,
+                30 => 2,
+                31 => 18,
+                i => bail!("cubby must be in the range 0-31, not {i}"),
+            };
+            debug!(log, "decoded cubby {c} => target {t}");
+            t
+        } else if let Some(s) = self.sidecar {
+            let t = match s {
+                SidecarSelector::Peer => 34,
+                SidecarSelector::Local => 35,
+            };
+            debug!(log, "decoded {s:?} => target {t}");
+            t
+        } else if let Some(p) = self.psc {
+            let t = match p {
+                0 => 32,
+                1 => 33,
+                i => bail!("psc must be in the range 0-1, not {i}"),
+            };
+            debug!(log, "decoded psc {p} => target {t}");
+            t
+        } else {
+            return Ok(None);
+        };
+        Ok(Some(t))
+    }
+}
+
+/// `IgnitionSelector` that also supports `--all`
+#[derive(Clone, Debug, clap::Args)]
+pub struct IgnitionBulkSelector {
+    #[clap(flatten)]
```

Comment:
Does this preclude having a `conficts_with_all = ["all"]`? Could we get that effect by making `IgnitionBulkSelector` an enum, or would that force us to have a subcommand to specify the variant?